webapp/src/actions/actions.ts

```diff
@@ -0,0 +1,126 @@
+import {AnyAction, Dispatch} from 'redux';
+
+import {getCurrentChannel} from 'mattermost-redux/selectors/entities/channels';
+import {getCurrentTeamId} from 'mattermost-redux/selectors/entities/teams';
+import {GetStateFunc} from 'mattermost-redux/types/actions';
+import {Client4} from 'mattermost-redux/client';
+import {IntegrationTypes} from 'mattermost-redux/action_types';
+
+import ActionTypes from 'action_types/';
+import {BadgeID} from 'types/badges';
+import {RHSState} from 'types/general';
+
+/**
+ * Stores`showRHSPlugin` action returned by
+ * registerRightHandSidebarComponent in plugin initialization.
+ */
+export function setShowRHSAction(showRHSPluginAction: () => void) {
+    return {
+        type: ActionTypes.RECEIVED_SHOW_RHS_ACTION,
+        showRHSPluginAction,
+    };
+}
+
+export function setRHSUser(userID: string | null) {
+    return {
+        type: ActionTypes.RECEIVED_RHS_USER,
+        data: userID,
+    };
+}
+
+export function setRHSBadge(badgeID: BadgeID | null) {
+    return {
+        type: ActionTypes.RECEIVED_RHS_BADGE,
+        data: badgeID,
+    };
+}
+
+export function setRHSView(view: RHSState) {
+    return {
+        type: ActionTypes.RECEIVED_RHS_VIEW,
+        data: view,
+    };
+}
+
+export function setTriggerId(triggerId: string) {
+    return {
+        type: IntegrationTypes.RECEIVED_DIALOG_TRIGGER_ID,
+        data: triggerId,
+    };
+}
+
+export function openGrant(user?: string, badge?: string) {
+    return (dispatch: Dispatch<AnyAction>, getState: GetStateFunc) => {
+        let command = '/badges grant';
+        if (user) {
+            command += ` --user ${user}`;
+        }
+
+        if (badge) {
+            command += ` --badge ${badge}`;
+        }
+
+        clientExecuteCommand(dispatch, getState, command);
```

Comment:
Really smart approach, reusing the whole server side command based dialogs to avoid repeating code here. Nice :)